src/particleContainer/ParticleContainer.h

```diff
@@ -170,6 +170,7 @@ class ParticleContainer: public MemoryProfilable {
 	//! This number may include particles which are outside of
 	//! the bounding box
 	virtual unsigned long getNumberOfParticles() = 0;
+	virtual unsigned long getNumberOfParticles(ParticleIterator::Type t) { return getNumberOfParticles(); }
```

Comment:
This seems misleading. If any argument is provided, disregard it and use the function that takes no argument?
I guess you meant to write:
```suggestion
	virtual unsigned long getNumberOfParticles() { return getNumberOfParticles(ParticleIterator::ALL_CELLS); }
	virtual unsigned long getNumberOfParticles(ParticleIterator::Type t) = 0;
```
?

Reply:
I simplified the readability of the default argument by using "real" default arguments. Hopefully that's easier and less error-prone.